app/scripts/map/cartodb-vis-directive.js

```diff
@@ -41,20 +42,32 @@
                 s.hide();
                 s.legend.set('visible', false);
             });
+            demographicsVisible = !!(sublayer);
             if (sublayer) {
                 sublayer.show();
                 sublayer.legend.set('visible', true);
             } else {
                 // Hide the sticky tooltip by clearing block styling...weee this is messy
-                $('div.cartodb-tooltip').css('display', '');
+                $('div.cartodb-tooltip .tooltip-tracts').parent().css('display', 'none');
             }
         }
 
         function onVisReady(vis) {
             map = vis.getNativeMap();
+            var pointsLayer = vis.getLayers()[1];
```

Comment:
Are you guaranteed to have your layer at that offset here? Might want to guard against array index out of range here otherwise.